cmd/root.go

```
@@ -1,61 +1,663 @@
+			}
+			found = true
+
+			updateComment(*ghComment, jComment, jIssue, ghClient, jClient)
+		}
+		if found {
+			continue
+		}
+
+		if err := createComment(*ghComment, jIssue, ghClient, jClient); err != nil {
+			return err
+		}
+	}
+
+	log.Debugf("Copied comments from GH issue #%d to JIRA issue %s.", *ghIssue.Number, jIssue.Key)
+	return nil
+}
+
+func updateComment(ghComment github.IssueComment, jComment jira.Comment, jIssue jira.Issue, ghClient github.Client, jClient jira.Client) error {
+	// fields[0] is the whole body, 1 is the ID, 2 is the username, 3 is the real name (or "" if none)
+	// 4 is the date, and 5 is the real body
+	fields := jCommentRegex.FindStringSubmatch(jComment.Body)
+
+	if fields[5] == *ghComment.Body {
+		return nil
+	}
+
+	user, _, err := ghClient.Users.Get(context.Background(), *ghComment.User.Login)
+	if err != nil {
+		log.Errorf("Error retrieving GitHub user %s. Error: %s", *ghComment.User.Login, err)
+	}
+	body := fmt.Sprintf("Comment (ID %d) from GitHub user %s", *ghComment.ID, user.GetLogin())
+	if user.GetName() != "" {
+		body = fmt.Sprintf(
+			"%s (%s) at %s:\n\n%s",
+			body,
+			user.GetName(),
+			ghComment.CreatedAt.Format(commentDateFormat),
+			*ghComment.Body,
+		)
+	} else {
+		body = fmt.Sprintf(
+			"%s at %s:\n\n%s",
+			body,
+			ghComment.CreatedAt.Format(commentDateFormat),
+			*ghComment.Body,
+		)
+	}
+
+	// As it is, the JIRA API we're using doesn't have any way to update comments natively.
+	// So, we have to build the request ourselves.
+
+	request := struct {
+		Body string `json:"body"`
+	}{
+		Body: body,
+	}
+
+	req, err := jClient.NewRequest("PUT", fmt.Sprintf("rest/api/2/issue/%s/comment/%s", jIssue.Key, jComment.ID), request)
+	if err != nil {
+		log.Errorf("Error creating comment update request: %s", err)
+		return err
+	}
+
+	res, err := jClient.Do(req, nil) // Response should be 204 No Content
+	if err != nil {
+		log.Errorf("Error updating comment: %s", err)
+		body, _ := ioutil.ReadAll(res.Body)
```

do not ignore the error returned by ReadAll; handle it as necessary